x-pack/plugins/spaces/server/lib/toggle_ui_capabilities.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { UICapabilities } from 'ui/capabilities';
+import { Space } from '../../common/model/space';
+import { SpacesClient } from './spaces_client';
+
+export async function toggleUiCapabilities(
+  uiCapabilities: UICapabilities,
+  activeSpace: Space,
+  spacesClient: SpacesClient
+) {
+  // 1) Determine if user can manage spaces
+  await toggleManageSpacesCapability(uiCapabilities, spacesClient);
+
+  // 2) Turn off capabilities that are disabled within this space
+  toggleDisabledFeatures(uiCapabilities, activeSpace);
+
+  return uiCapabilities;
+}
+
+async function toggleManageSpacesCapability(
+  uiCapabilities: UICapabilities,
+  spacesClient: SpacesClient
+) {
+  // Spaces special case
```

Comment:
I *think* we should be able to add a "UI" capability action to the privileges which let you manage spaces so we don't need this special check here.

Reply:
I think you're right!